packages/js/admin-e2e-tests/package.json

```diff
@@ -50,25 +67,45 @@
 		"jest-mock-extended": "^1.0.18",
 		"rimraf": "^3.0.2",
 		"ts-jest": "~29.1.1",
-		"typescript": "^5.3.3"
+		"typescript": "^5.3.3",
+		"wireit": "^0.14.1"
 	},
 	"publishConfig": {
 		"access": "public"
 	},
-	"scripts": {
-		"turbo:build": "tsc --project tsconfig.json",
-		"prepare": "composer install",
-		"changelog": "composer exec -- changelogger",
-		"build": "pnpm -w exec turbo run turbo:build --filter=$npm_package_name",
-		"lint": "eslint src",
-		"start": "tsc --project tsconfig.json --watch",
-		"clean": "pnpm exec rimraf tsconfig.tsbuildinfo build build-*",
-		"lint:fix": "eslint src --fix",
-		"prepack": "pnpm run clean && pnpm run build"
-	},
-	"lint-staged": {
-		"*.(t|j)s?(x)": [
-			"pnpm lint:fix"
-		]
+	"wireit": {
+		"build:project:typescript": {
+			"command": "tsc --project tsconfig.json",
+			"files": [
+				"tsconfig.json",
+				"src/**/*.{js,jsx,ts,tsx}",
+				"typings/**/*.ts"
+			],
+			"output": [
+				"build"
+			],
+			"dependencies": [
+				"dependencyOutputs"
+			]
+		},
+		"dependencyOutputs": {
```

Comment:
Wireit supports [basic globs](https://github.com/google/wireit#glob-patterns), so naively is there a reason we couldn't use globs to match eslint plugin and friends, and at least reduce the amount of negative matchers we need?

`allowUsuallyExcludedPaths` lets you mention a usually excluded path, so in my mind the behaviour should be, if i match like `node_modules/eslint-plugin/*.js` it wouldn't auto include the usual excluded files like `.git`, it just lets me match on JS files under `node_modules`. Is that not how it behaves?

Reply:
Notably, we only fall back to the aggressive include/exclude [when the package has no files defined](https://github.com/woocommerce/woocommerce/blob/d06cd2f7298275c306f8c4c3d7cb29224534b2a8/.pnpmfile.cjs#L73-L106). I went with the heavy-handed approach for the generic inclusion because we should consider any "output" file for the fingerprinting.

For example, if we only used a glob for JS files, changes to PHP files wouldn't invalidate the cache. We could try using globs for more file types but since things like CSV and JSON files can impact the build, at some point, we may as well have had a glob for the entire directory.

What we _could_ do is go through all of the private packages (and plugin packages) and make sure they define a `files` glob that encompasses the files that impact the build output. Since we don't distribute them there's no reason we can't include PHP files in that "package" too. Personally, I like this idea, but maybe we can visit that in a follow-up PR?

Reply:
@ObliviousHarmony yeah ok I see your point here. Fine with me to be exhaustive and look at improvements later 👍🏻